[Improve][Manager] Update base64 encoder

### Description

Now the manager encodes the token with base-64 encoder from `sun.misc.BASE64Encoder` in `org.apache.inlong.manager.service.resource.sink.es.ElasticsearchApi`
```java
String token = String.valueOf(new BASE64Encoder().encode(tokenStr.getBytes(StandardCharsets.UTF_8)));
```
But `sun.misc.BASE64Encoder` has been removed from JDK9, so there will be an error when compiling InLong project with JDK after version 9.

```shell
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /data1/tools/code/inlong/inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java:[42,16] cannot find symbol
  symbol:   class BASE64Encoder
  location: package sun.misc
```

Actually, JDK8 provides an alternative API to enable base64 encoder and decoder. https://docs.oracle.com/javase/9/docs/api/java/util/Base64.Encoder.html
https://docs.oracle.com/javase/9/docs/api/java/util/Base64.Decoder.html

So, it's better to refactor the base64 encoder and decoder with the new APIs.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java
+++ b/inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java
@@ -17,8 +17,6 @@
 
 package org.apache.inlong.manager.service.resource.sink.es;
 
-import java.util.Base64;
-import java.util.Base64.Encoder;
 import org.apache.inlong.manager.common.consts.InlongConstants;
 import org.apache.inlong.manager.common.util.HttpUtils;
 import org.apache.inlong.manager.pojo.sink.es.ElasticsearchCreateIndexResponse;
@@ -45,6 +43,8 @@ import org.springframework.web.client.HttpClientErrorException.NotFound;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Base64.Encoder;
 import java.util.List;
 import java.util.Map;
 
@@ -70,7 +70,6 @@ public class ElasticsearchApi {
     private static final String FIELD_FORMAT = "format";
 
     private static final String CONTENT_TYPE_VALUE = "application/json;charset=UTF-8";
-
 
     private final Encoder base64Encoder = Base64.getEncoder();
 

--- a/inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java
+++ b/inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java
@@ -17,6 +17,8 @@
 
 package org.apache.inlong.manager.service.resource.sink.es;
 
+import java.util.Base64;
+import java.util.Base64.Encoder;
 import org.apache.inlong.manager.common.consts.InlongConstants;
 import org.apache.inlong.manager.common.util.HttpUtils;
 import org.apache.inlong.manager.pojo.sink.es.ElasticsearchCreateIndexResponse;
@@ -39,7 +41,6 @@ import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
 import org.springframework.stereotype.Component;
 import org.springframework.web.client.HttpClientErrorException.NotFound;
-import sun.misc.BASE64Encoder;
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
@@ -53,6 +54,8 @@ import java.util.Map;
 @Slf4j
 @Component
 public class ElasticsearchApi {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchApi.class);
 
     private static final Gson GSON = new GsonBuilder().create();
 
@@ -68,7 +71,8 @@ public class ElasticsearchApi {
 
     private static final String CONTENT_TYPE_VALUE = "application/json;charset=UTF-8";
 
-    private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchApi.class);
+
+    private final Encoder base64Encoder = Base64.getEncoder();
 
     @Autowired
     private ElasticsearchConfig esConfig;
@@ -84,7 +88,7 @@ public class ElasticsearchApi {
         if (esConfig.getAuthEnable()) {
             if (StringUtils.isNotEmpty(esConfig.getUsername()) && StringUtils.isNotEmpty(esConfig.getPassword())) {
                 String tokenStr = esConfig.getUsername() + ":" + esConfig.getPassword();
-                String token = String.valueOf(new BASE64Encoder().encode(tokenStr.getBytes(StandardCharsets.UTF_8)));
+                String token = base64Encoder.encodeToString(tokenStr.getBytes(StandardCharsets.UTF_8));
                 headers.add("Authorization", "Basic " + token);
             }
         }
@@ -214,6 +218,7 @@ public class ElasticsearchApi {
         JsonObject fields = null;
         Map<String, ElasticsearchFieldInfo> fieldInfos = Maps.newHashMap();
         if (ObjectUtils.isNotEmpty(mappings)) {
+            String MAPPINGS_KEY = "mappings";
             properties = mappings.getAsJsonObject(MAPPINGS_KEY);
         }
         if (ObjectUtils.isNotEmpty(properties)) {

--- a/inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/resource/sink/Base64CompatibleTest.java
+++ b/inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/resource/sink/Base64CompatibleTest.java
@@ -1,0 +1,24 @@
+package org.apache.inlong.manager.service.resource.sink;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Base64.Encoder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import sun.misc.BASE64Encoder;
+
+public class Base64CompatibleTest {
+
+    @Test
+    public void base64CompatibleForDifferentVersionTest() {
+        String tokenStr = "root" + ":" + "admin";
+        byte[] tokeBytes = tokenStr.getBytes(StandardCharsets.UTF_8);
+        // token encode by sun.misc.BASE64Encoder, this test will fail if the jdk version is 9 or later
+        // so, this test should be removed when Apache InLong project support jdk 9 or later
+        String tokenOld = String.valueOf(new BASE64Encoder().encode(tokeBytes));
+        // token encode by java.util.Base64.Encoder
+        Encoder encoder = Base64.getEncoder();
+        String tokenNew = encoder.encodeToString(tokeBytes);
+        Assertions.assertEquals(tokenOld, tokenNew);
+    }
+}

--- a/inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/resource/sink/Base64CompatibleTest.java
+++ b/inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/resource/sink/Base64CompatibleTest.java
@@ -1,11 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.apache.inlong.manager.service.resource.sink;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import sun.misc.BASE64Encoder;
 
 import java.nio.charset.StandardCharsets;
 import java.util.Base64;
 import java.util.Base64.Encoder;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Test;
-import sun.misc.BASE64Encoder;
 
 public class Base64CompatibleTest {
 

